### PR TITLE
Add 'key' meta info field to INavLink interface

### PR DESCRIPTION
**Pull request by inateeg**
add 'id' optional field to INavLink so that when getting data from varies dataSource, we persist the unique id with link if applies.

### CODE REVIEW

File: src/components/Nav/Nav.tsx [hunk 1]
```diff
@@ -51,7 +51,6 @@ export class Nav extends React.Component<INavProps, INavState> {
           className={'ms-Nav-link' + (_isLinkSelected(link) ? ' is-selected' : '')}
           href={ link.url || 'javascript:' }
           onClick={ onLinkClick }
-          title={ link.title ? link.title : undefined }
```
Comment (cschlechty) [inline src/components/Nav/Nav.tsx]: Why is this change removing the title?

Comment (inateeg) [inline src/components/Nav/Nav.tsx]: using ariaLabel instead

Comment (cschlechty) [inline src/components/Nav/Nav.tsx]: Those have different purposes. Title is for tooltips which may be desired (and possibly already in use) by others.

Comment (inateeg) [inline src/components/Nav/Nav.tsx]: OK add a tooltip field and render title if tooltip is set.

File: src/components/Nav/Nav.Props.ts [hunk 2]
```diff
@@ -78,17 +78,21 @@ export interface INavLink {
...
    */
-  ariaLabel?: string;
+  id?: string;
 
   /**
-   * (Optional) Title for nav link
+   * (Optional) Desciption
    */
-  title?: string;
+  tooltip?: string;
```
Comment (dzearing) [inline src/components/Nav/Nav.Props.ts]: this should stay as "title". It is confusing to rename the standard "title" attribute to "tooltip".

File: src/components/Nav/Nav.Props.ts [hunk 3]
```diff
@@ -78,17 +78,21 @@ export interface INavLink {
...
    */
-  [propertyName: string]: any;
+  ariaLabel?: string;
 
   /**
-   * (Optional) Aria label for nav link
+   * (Optional) Unique ID for nav link
    */
-  ariaLabel?: string;
+  id?: string;
```
Comment (dzearing) [inline src/components/Nav/Nav.Props.ts]: I am super sorry for suggesting "id" but I think you should rename this back to "key". I didn't realize you were talking about INavLink, thought you mentioned a component's props interface. "key" should not be used in props interfaces, but "key" can and probably should be used in object interfaces. It describes the unique value for the object that should be used when rendering the ui representation of the object.

Comment (dzearing) [inline src/components/Nav/Nav.Props.ts]: Also I don't see where you're actually using it.

**Comment by dzearing** [comment]
Please edit your PR title to be more descriptive of your change. "Nav1" is not a very descriptive PR title. :)

File: src/components/Nav/Nav.Props.ts (hunk 3 above)
Comment (inateeg) [inline src/components/Nav/Nav.Props.ts]: OK revert back to key  this is meta info field needed to store the persisted key for each link during operations.

File: src/components/Nav/Nav.Props.ts [hunk 4]
```diff
@@ -78,17 +78,21 @@ export interface INavLink {
...
 
   /**
-   * (Optional) Title for nav link
+   * (Optional) title for tooltip or description
    */
   title?: string;
+  /**
+   * Any additional properties to apply to the rendered links.
+   */
+  [propertyName: string]: any;
```
Comment (dzearing) [inline src/components/Nav/Nav.Props.ts]: this is ambiguous and not sure where it's being used. I think if you intend to add attributes to the link, we should call it 

```
linkProps?: React.HTMLProps<HTMLAnchorElement>
```

Comment (inateeg) [inline src/components/Nav/Nav.Props.ts]: I thought this field [propertyName: string]: any always there ?  Unless someone just removed it?  I have not sync up to the latest?  I don't need this and I think this was originally added from day one to mimic the ILeftNavItem customListProperties.
I can remove it if you think that is right thing to do.

Comment (inateeg) [inline src/components/Nav/Nav.Props.ts]: Found out this property always there if you see your initial copy.  But it is not at the end.  I only moved it to the end as standard way.

Comment (dzearing) [inline src/components/Nav/Nav.Props.ts]: i see.

**Comment by dzearing** [comment]
Approved

**Comment by inateeg** [comment]
David, do we know whe office react will be published tonight?

**Pull request closed without merging**